packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/useFileUploader.tsx

```diff
@@ -8,25 +8,33 @@ export default function useFileUploader(): UseFileUploader {
 
   const [inDropZone, setInDropZone] = useState(false);
 
-  const onDragStart = (e: React.DragEvent<HTMLDivElement>) => {
-    e.dataTransfer.clearData();
-    // stubbed
+  const onDragStart = (event: React.DragEvent<HTMLDivElement>) => {
+    event.dataTransfer.clearData();
   };
-  const onDragEnter = (e: React.DragEvent<HTMLDivElement>) => {
-    e.preventDefault();
-    // stubbed
+  const onDragEnter = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
   };
-  const onDragLeave = (e: React.DragEvent<HTMLDivElement>) => {
-    e.preventDefault();
-    // stubbed
+  const onDragLeave = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setInDropZone(false);
   };
-  const onDragOver = (e: React.DragEvent<HTMLDivElement>) => {
-    e.preventDefault();
-    // stubbed
+  const onDragOver = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setInDropZone(true);
+    event.dataTransfer.dropEffect = 'copy';
   };
-  const onDrop = (e: React.DragEvent<HTMLDivElement>) => {
-    e.preventDefault();
-    //stubbed
+  const onDrop = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    const files = [...event.dataTransfer.files];
```

Comment:
Why are we spreading here? Seems extraneous

Reply:
The `event.dataTransfer.files` is a `FileList` , which is not an array of files. So to convert it to an array of Files[], which will be easier to deal with, I converted it over. https://stackoverflow.com/questions/25333488/why-isnt-the-filelist-object-an-array